server/src/main/java/org/opensearch/node/remotestore/RemoteStorePinnedTimestampService.java

```diff
@@ -147,8 +126,8 @@ private void startAsyncUpdateTask() {
     public void pinTimestamp(long timestamp, String pinningEntity, ActionListener<Void> listener) {
         // If a caller uses current system time to pin the timestamp, following check will almost always fail.
         // So, we allow pinning timestamp in the past upto some buffer
-        long lookbackIntervalInMills = CLUSTER_REMOTE_STORE_PINNED_TIMESTAMP_LOOKBACK_INTERVAL.get(settings).millis();
-        if (timestamp < TimeUnit.NANOSECONDS.toMillis(System.nanoTime()) - lookbackIntervalInMills) {
+        long lookbackIntervalInMills = remoteStoreSettings.getPinnedTimestampsLookbackInterval().millis();
+        if (timestamp < (System.currentTimeMillis() - lookbackIntervalInMills)) {
```

Comment:
Sorry why do we need to support any time in the past, while future intervals should be fine

Reply:
This is to handle scenario where the time since caller fetched the value of now and calling `pinTimestamp()`